java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java

```diff
@@ -105,6 +106,23 @@ public void loadFieldBuffers(ArrowFieldNode fieldNode, List<ArrowBuf> ownBuffers
   }
 
   @Override
+  public void loadFieldBuffers(BuffersIterator buffersIterator, ArrowBuf buf) {
+    buffersIterator.next();
+    ArrowBuf offsetsData = buf.slice((int) buffersIterator.offset(), (int) buffersIterator.length());
+    offsets.load(offsetsData);
+    if (buffersIterator.length() == 0) {
+      offsets.allocateNew(1);
+      offsets.zeroVector();
+    }
+    lastSet = (int) buffersIterator.length() / 4 - 1;
+    // this doesn't match arrow spec, but for backward compatibility, doing it in this order to match getBuffers() method
```

Comment:
In what way is it out of spec (the integration tests pass on this patch, for example)?